src/app/lib/access/logger/logger_repository.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+/// Contract repository to handle logging configuration.
+abstract interface class LoggerRepository {
+  factory LoggerRepository() = _LoggerRepository;
+
+  /// Loads the current logging configuration
+  Future<LoggingConfiguration> getLoggingConfiguration();
+
+  /// Sets whether console logging should be enabled on next app launch.
+  Future<bool> setIsConsoleLoggingEnabled(bool isConsoleLoggingEnabled);
+
+  /// Sets whether file logging should be enabled on next app launch.
+  Future<bool> setIsFileLoggingEnabled(bool isFileLoggingEnabled);
+}
+
+final class LoggingConfiguration {
+  final bool? isConsoleLoggingEnabled;
+  final bool? isFileLoggingEnabled;
+
+  const LoggingConfiguration({
+    required this.isConsoleLoggingEnabled,
+    required this.isFileLoggingEnabled,
+  });
+}
+
+/// Implementation of [LoggerRepository].
+final class _LoggerRepository implements LoggerRepository {
+  /// The key used to store the selected environment in shared preferences.
+  final String _consoleLoggingKey = 'consoleLogging';
+  final String _fileloggingKey = 'fileLogging';
+
+  @override
+  Future<LoggingConfiguration> getLoggingConfiguration() async {
```

Comment:
Could this be ever be called with no persisted value?

Reply:
yeah, `LoggingConfiguration` properties are nullable. Therefore, if one of them is not persisted, it will be null instead of `true` or `false`